lib/SIL/SILBasicBlock.cpp

```diff
@@ -314,17 +314,34 @@ ScopeCloner::getOrCreateClonedScope(const SILDebugScope *OrigScope) {
   if (it != ClonedScopeCache.end())
     return it->second;
 
-  auto ClonedScope = new (NewFn.getModule()) SILDebugScope(*OrigScope);
-  if (OrigScope->InlinedCallSite) {
+  const SILDebugScope *ClonedInlinedCallSite = nullptr;
+  SILFunction *ClonedFn;
```

Comment:
`= nullptr;` and then only set it when necesssary?

Reply:
I had that before and it led me to a bug. It's set exactly once on every branch this way, with every branch different.